optuna/samplers/_cmaes.py

```diff
@@ -352,38 +358,35 @@ def _restore_optimizer(
 
     def _init_optimizer(
         self,
-        search_space: Dict[str, BaseDistribution],
-        ordered_keys: List[str],
+        trans: _SearchSpaceTransform,
         population_size: Optional[int] = None,
         randomize_start_point: bool = False,
     ) -> CMA:
+
+        bds = trans.bounds
+
         if randomize_start_point:
             # `_initialize_x0_randomly ` returns internal representations.
-            x0 = _initialize_x0_randomly(self._cma_rng, search_space)
-            mean = np.array([x0[k] for k in ordered_keys], dtype=float)
+            mean = np.array([b[0] + (b[1] - b[0]) * self._cma_rng.rand() for b in bds])
         elif self._x0 is None:
             # `_initialize_x0` returns internal representations.
-            x0 = _initialize_x0(search_space)
-            mean = np.array([x0[k] for k in ordered_keys], dtype=float)
+            mean = np.array([b[0] + (b[1] - b[0]) / 2 for b in bds])
```

Comment:
How about calculating a mean vector like the following to remove list comprehensions?

```python
lower_bounds = trans.bounds[:, 0]
upper_bounds = trans.bounds[:, 1]
n_dimension = len(trans.bounds)

if randomize_start_point:
    mean = lower_bounds + (upper_bounds - lower_bounds) * self._cma_rng.rand(n_dimension)
elif self._x0 is None:
    mean = lower_bounds + (upper_bounds - lower_bounds) / 2
```

Reply:
Done. Thanks!